yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnClientSchedulerBackend.scala

```diff
@@ -64,7 +68,7 @@ private[spark] class YarnClientSchedulerBackend(
     // SPARK-8851: In yarn-client mode, the AM still does the credentials refresh. The driver
     // reads the credentials from HDFS, just like the executors and updates its own credentials
     // cache.
-    if (conf.contains("spark.yarn.credentials.file")) {
+    if (!conf.contains(PRINCIPAL.key) && conf.contains("spark.yarn.credentials.file")) {
```

Comment:
That's a different issue, and I'm not sure it's really an issue, given what the comment above this line says.

This code is here for when Spark is managing your kerberos credentials and delegation tokens, i.e., when you pass `--principal` and `--keytab`. In that case, as the comment above says, the AM is tasked with renewing the credentials and tokens periodically, and this code exists so that the driver (which is not running in the AM in client mode) can get the new tokens.

You're basically breaking that feature by changing this. If your app is managing the kerberos login, you'd never pass `--principal` and `--keytab` (or the equivalent settings) to Spark, so you wouldn't run into this problem.